text/0000-tuple-struct-self-ctor.md

```diff
@@ -0,0 +1,198 @@
+- Feature Name: tuple_struct_self_ctor
+- Start Date: 2017-01-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Tuple `struct`s can now be constructed with `Self(v1, v2, ..)`
+to match how `struct`s with named fields can be constructed
+using `Self { f1: v1, f2: v2, .. }`. A simple example:
+
+```rust
+struct TheAnswer(usize);
+
+impl Default for TheAnswer {
+    fn default() -> Self { Self(42) }
+}
+```
```

Comment:
If tuple structs are supported, then unit structs can be supported in the same way:
```rust
struct TheAnswer;

impl Default for TheAnswer {
    fn default() -> Self { Self }
}
```

Reply:
I considered it initially but didn't for some reason I can't remember. I can add that to the RFC if you like.

Reply:
There are three forms of instantiation (to use the word slightly fuzzily): `Self { }`, `Self()` and `Self`. `Self { }` already works. You’ve proposed adding `Self()` for consistency. It doesn’t seem to me that it fixes consistency properly unless `Self` is also made to work.

Reply:
@chris-morgan Sure, I'll change the RFC accordingly Soon™.